pydis_site/apps/content/resources/guides/pydis-guides/contributing/bot.md

```diff
@@ -2,207 +2,643 @@
-| `REDIS_PASSWORD` | When not using FakeRedis | The password to connect to the redis database. *Leave empty if you're not using REDIS.* |
-
----
+These are the main sections in `config-default.yml` that need overriding:
 
-If you are running on the host, while not required, we advise you set `use_fakeredis` to `true` in your `config.yml` file during development to avoid the need of setting up a Redis server.
-It does mean you may lose persistent data on restart but this is non-critical.
-Otherwise, you should set up a Redis instance and fill in the necessary config.
-{: .notification .is-warning }
+* `guild.id`
+* `guild.categories`
+* `guild.channels`
+* `guild.roles`
+* `guild.webhooks`
+* `style.emojis`
 
----
+Additionally:
 
-Example `.env` file:
+* Set `urls.site_schema` and `urls.site_api_schema` to `"http://"`.
+* Set `urls.site_api` to `!JOIN ["api.", *DOMAIN]`.
+* Set `urls.snekbox_eval_api` to `"http://localhost:8060/eval"`.
+* At this stage, set `bot.redis.use_fakeredis` to `true`. If you're looking for instructions for working with Redis, see [Working with Redis](#optional-working-with-redis).
 
-```shell
-BOT_TOKEN=YourDiscordBotTokenHere
-BOT_API_KEY=badbot13m0n8f570f942013fc818f234916ca531
-REDDIT_CLIENT_ID=YourRedditClientIDHere
-REDDIT_SECRET=YourRedditSecretHere
-```
+We understand this is tedious and are working on a better solution for setting up test servers.
 
----
-# Run the project
+<div class="card">
+    <button type="button" class="card-header collapsible">
+        <span class="card-header-title subtitle is-6 my-2 ml-2">Why do you need a separate config file?</span>
+        <span class="card-header-icon">
+            <i class="fas fa-angle-down title is-5" aria-hidden="true"></i>
+        </span>
+    </button>
+    <div class="collapsible-content">
+        <div class="card-content">
+            While it's technically possible to edit <code>config-default.yml</code> to match your server, it is heavily discouraged.
+            This file's purpose is to provide the configurations the Python bot needs to run in the Python server in production, and should remain as such.
+            In contrast, the <code>config.yml</code> file can remain in your local copy of the code, and will be ignored by commits via the project's <code>.gitignore</code>.
+        </div>
+    </div>
+</div>
+<br>
+
+#### .env
+The second file you need to open is the one containing the environment variables, and needs to have the name `.env`. 
+Inside, add the line `BOT_TOKEN=YourDiscordBotTokenHere`. See [here](../creating-bot-account) for help with obtaining the bot token.
+
+The `.env` file will be ignored by commits.
+
+### Run it!
+#### With Docker
+You are now almost ready to run the Python bot. The simplest way to do so is with Docker. 
+
+In your `config.yml` file, set `urls.site` to `"web:8000"`.
```

Comment:
So, I think we should be default assume people are using Docker and move this up to the previous section talking about config.yml. It's a bit odd to think we're done with config.yml only to have it randomly pop up here again.

I think when looking at the running the bot locally, you can specify that the config.yml has to be different than specified above.

Reply:
Yeah that's fair